### PR TITLE
clang/AMDGPU: Emit atomicrmw for global/flat fadd v2bf16 builtins

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96875?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#96876** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96876?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96875** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96875?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#96874** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96874?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96873** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96873?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96872** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96872?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96760** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96760?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96759** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96759?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96444** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96444?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96443** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96442** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96442?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95930** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95930?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95929** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95929?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by arsenm** [comment]
ping

File: clang/test/CodeGenOpenCL/builtins-fp-atomics-gfx12.cl [hunk 1]
```diff
@@ -48,7 +48,7 @@ void test_local_add_2f16_noret(__local half2 *addr, half2 x) {
 }
 
 // CHECK-LABEL: test_flat_add_2f16
-// CHECK: [[RMW:%.+]] = atomicrmw fadd ptr %{{.+}}, <2 x half> %{{.+}} syncscope("agent") seq_cst, align 4, !amdgpu.no.fine.grained.memory !{{[0-9]+$}}
+// CHECK: [[RMW:%.+]] = atomicrmw fadd ptr %{{.+}}, <2 x half> %{{.+}} syncscope("agent") monotonic, align 4, !amdgpu.no.fine.grained.memory !{{[0-9]+$}}
```
Comment (shiltian) [inline clang/test/CodeGenOpenCL/builtins-fp-atomics-gfx12.cl:51]: why was this changed?

**Review by shiltian — APPROVED** [review]
LGTM with one question regarding the memory order

File: clang/test/CodeGenOpenCL/builtins-fp-atomics-gfx12.cl (hunk 1 above)
Comment (arsenm) [reply clang/test/CodeGenOpenCL/builtins-fp-atomics-gfx12.cl:51]: I'm not sure. My guess is this changed in the downstream patch change to seq_cst to monotonic, and this diff wasn't updated? Before this patch the atomicrmw wouldn't have been produced at all

Comment (shiltian) [reply clang/test/CodeGenOpenCL/builtins-fp-atomics-gfx12.cl:51]: it passes BuildBot tests

**Comment by arsenm** [comment]
### Merge activity

* **Aug 20, 2:53 PM EDT**: @arsenm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/96875).
